docs/configuring_and_running_tests.md

```diff
@@ -0,0 +1,493 @@
+a trace (only the primary reference implementations are instrumented to capture such traces).
+
+### Selecting Test Cases
+
+The `connectconformance` test runner supports four different options for selecting which test cases to
+run. Two of them are intended for interactive runs and the other two are suggested for use with test
+runs during continuous integration testing.
+
+1. `--known-failing`: The most likely option you may need to use is to mark tests that are known to
+   fail. When an implementation is not 100% compliant, this can be used to opt out of the test cases
+   that do not pass. With this flag, these test cases will be run, but expected to fail. In fact, if
+   they happen to pass, the test run will be considered a failure since it means that the known
+   failing configuration is stale and needs to be updated to match recent fixes.
+2. `--known-flaky`: This is similar to above, but is for tests that do not consistently fail. This
+   is often indicative of timing-related and concurrency bugs, where the sequence of execution can
+   be non-deterministic, so sometimes a test passes and sometimes it fails. In this mode, the test
+   cases are still run, but allowed to fail. Whether the test case passes or fails does not cause
+   the whole test run to pass or fail. But if it does fail, it will be logged in the test output.
+3. `--run`: This option is intended for interactive runs, like when troubleshooting particular
+   test cases. Instead of running the entire suite, you can run just select test cases.
+4. `--skip`: This option is also intended for interactive runs. It is the  opposite of `--run`
+   and allows you to skip execution of certain test cases. When combined with `--run`, you can
+   use the `--run` flag to enumerate which cases are considered, and then `--skip` refines
+   the set, skipping some of the test cases that otherwise match a `--run` pattern.
+
+All four of these options accept two kinds of values:
+1. A _test case pattern_. This is like a full test case name but it can have wildcards. An
+   asterisk (`*`) matches one name component. A double-asterisk (`**`) matches zero or more
+   components. A name component is the portion of the name between `/` separators. Wildcards
+   can**not** be used to match partial name components. For example `foo/bar*` is not valid
+   since the wildcard is expected to only match a suffix. If such a patter is used, the
+   asterisk is matched exactly instead of being treated as a wildcard.
+2. If prefixed with an at-sign (`@`), the rest of the value is the path of a file that
+   contains test case patterns, one per line. When the file is processed, leading and trailing
+   whitespace is discarded from each line, blank lines are ignored, and lines that start with
+   a pound-sign (`#`) are treated as comments and ignored.
+
+All four of these options can be provided multiple times on the command-line, to provide
```

Comment:
The Linear task mentions:

> (and we should advise that the first two should be strongly preferring in CI environments; the latter two mostly for exploratory testing or troubleshooting particular test cases).

I know you touched on it above, but you may want to stress that again here explicitly since it may not be entirely clear.